tests/test_benchmark_stratified_thermal_storage.py

```diff
@@ -0,0 +1,836 @@
+
+"""
+
+import argparse
+import os
+import shutil
+import pandas as pd
+
+import pytest
+import mock
+
+from multi_vector_simulator.cli import main
+
+from _constants import (
+    EXECUTE_TESTS_ON,
+    TESTS_ON_MASTER,
+    TEST_REPO_PATH,
+    CSV_EXT,
+)
+
+from multi_vector_simulator.utils.constants_json_strings import (
+    INSTALLED_CAP,
+    THERM_LOSSES_REL,
+    THERM_LOSSES_ABS,
+)
+
+TEST_INPUT_PATH = os.path.join(
+    TEST_REPO_PATH, "benchmark_test_inputs", "Feature_stratified_thermal_storage"
+)
+TEST_OUTPUT_PATH = os.path.join(TEST_REPO_PATH, "benchmark_test_outputs")
+
+
+class TestStratifiedThermalStorag:
+    def setup_method(self):
+        if os.path.exists(TEST_OUTPUT_PATH):
+            shutil.rmtree(TEST_OUTPUT_PATH, ignore_errors=True)
+        if os.path.exists(TEST_OUTPUT_PATH) is False:
+            os.mkdir(TEST_OUTPUT_PATH)
+
+        self.inputs_csv = os.path.join(TEST_INPUT_PATH, "csv_elements")
+        self.storage_csv = os.path.join(self.inputs_csv, "energyStorage.csv")
+
+        self.storage_fix_without_fixed_losses = (
+            "storage_fix_without_fixed_thermal_losses.csv"
+        )
+        self.storage_opt_without_fixed_losses = (
+            "storage_optimize_without_fixed_thermal_losses.csv"
+        )
+        self.storage_opt_with_fixed_losses_float = (
+            "storage_optimize_with_fixed_thermal_losses_float.csv"
+        )
+        self.storage_opt_with_fixed_losses_series = (
+            "storage_optimize_with_fixed_thermal_losses_time_series.csv"
+        )
+        self.storage_opt_with_zero_fixed_losses_float = (
+            "storage_optimize_with_zero_fixed_thermal_losses_float.csv"
+        )
+        self.storage_opt_with_zero_fixed_losses_series = (
+            "storage_optimize_with_zero_fixed_thermal_losses_time_series.csv"
+        )
+
+        self.storage_xx = os.path.join(
+            self.inputs_csv, self.storage_opt_with_fixed_losses_float
+        )
+
+        self.D1 = os.path.abspath(
+            os.path.join(
+                TEST_REPO_PATH,
+                os.pardir,
+                "src",
+                "multi_vector_simulator",
+                "D1_model_components.py",
+            )
+        )
+        self.fixed_losses = [THERM_LOSSES_REL, THERM_LOSSES_ABS]
+
+    # # this ensure that the test is only ran if explicitly executed, ie not when the `pytest` command
+    # # alone is called
+    @pytest.mark.skipif(
+        EXECUTE_TESTS_ON not in (TESTS_ON_MASTER),
+        reason="Benchmark test deactivated, set env variable "
+        "EXECUTE_TESTS_ON to 'master' to run this test",
+    )
+    @mock.patch("argparse.ArgumentParser.parse_args", return_value=argparse.Namespace())
+    def test_fix_generic_storage_with_default_losses(self, margs):
+        r"""
+        This test checks if the fix GenericStorage matches the one with additional
+        fixed thermal losses: fixed_thermal_losses_relative and
+        fixed_thermal_losses_absolute.
```

Comment:
I don't understand why this test is necessary, but I am also quite far away from this topic right now. @smartie2076 did you review these benchmark tests?

Reply:
The background is that so far `fixed_thermal_losses_absolute` or `fixed_thermal_losses_relative` didn't exist in mvs. They were not passed with `GenericStorage` component in ` D1_model_components.py`. Now that I inserted them, @smartie2076 wanted to be sure, that this modification has no effect on the previous state in mvs. (Btw: We call the storage of this previous state  `GenericStorage` even if the stratified thermal storage is a `GenericStorage` as well.)

So I wrote this test, in which I run the simulation twice and compare their `timeseries_all_busses.xlsx`. In the first run I comment `fixed_thermal_losses_absolute` or `fixed_thermal_losses_relative` out in ` D1_model_components.py` and thus fake the previous state - the so called `GenericStorage`. In the second run I run the "regular" simulation on this branch with  `fixed_thermal_losses_absolute` or `fixed_thermal_losses_relative`.

Martha and I discussed it [here](https://github.com/rl-institut/multi-vector-simulator/pull/718#issuecomment-756205972) and [here](https://github.com/rl-institut/multi-vector-simulator/pull/718#issuecomment-756291641) in this PR.

I hope it became clearer. :)

Reply:
Yes, it did! Thank you for the explanations :)